builtin/plugins/dposv3/dpos.go

```diff
@@ -1837,6 +1839,11 @@ func slashValidatorDelegations(
 			if err := cachedDelegations.SetDelegation(ctx, delegation); err != nil {
 				return err
 			}
+			if err := emitSlashDelegationEvent(
+				ctx, delegation.Validator, delegation.Amount, &types.BigUInt{Value: toSlash}, statistic.SlashPercentage, false,
```

Comment:
Should the event also store the delegator address?

Reply:
Unnecessary, the delegator address of slashed delegation is always the validator address.

Reply:
ok I misunderstood the slashing mechanism. I will emit the delegation objects as we also need complete information to identify slashed delegations.